bodhi/messages/schemas/base.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (C) 2018-2019 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""
+Message schema for Bodhi.
+
+Each message is defined as a Python class. For details, see `fedora-messaging
+<https://fedora-messaging.readthedocs.io/en/stable/>`_ documentation on
+messages.
+"""
+
+import re
+import typing
+
+from fedora_messaging import message
+from fedora_messaging.schema_utils import user_avatar_url
+
+
+class BodhiMessage(message.Message):
+    """A base class for Bodhi messages."""
+
+    def _title_or_alias(self):
+        value = self._body['update'].get('alias')
+        if not value:
+            value = self._body['update']['title']
+        return value
+
+    @property
+    def app_icon(self) -> str:
+        """
+        Return a URL that points to the application's icon.
+
+        This is used when displaying the message to users.
+
+        Returns:
+            A URL for Bodhi's app icon.
+        """
+        return "https://apps.fedoraproject.org/img/icons/bodhi.png"
+
+    def _get_agent_username(self):
+        """Return the agent's username for this message.
+
+        Returns:
+            str or None: The agent's username.
+        """
+        username = self._body.get('agent', None)
+        if isinstance(username, dict):
+            username = username['name']
```

Comment:
@abompard this pattern above is found in several of the message classes and is causing coverage issues because the test messages don't generally have users expressed this way. This code was copied from the old fedmsg code, right? Is it likely that the messages were once this way, and so this code was there for a transition long ago, and Bodhi doesn't encode usernames this way anymore? How complete are the set of messages in the tests? Do they include one of every type of message that Bodhi sends today?

Reply:
Yeah it was copied from old fedmsg code, you can very probably drop it, those messages may not be going around on the bus anymore.

Reply:
OK - I'm currently going through the tests and making them be consistent with the data structures I see in datagrepper. I am dropping the tests that check "legacy" data.